ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,70 @@ class RealtimeClientChannel: QuickSpec {
+                            options.autoConnect = false
+                            let client = ARTRealtime(options: options)
+                            client.setTransportClass(TestProxyTransport.self)
+                            client.connect()
+                            defer { client.close() }
+
+                            let (keyData, ivData, messages) = AblyTests.loadCryptoTestData(cryptoTest)
+                            let testMessage = messages[0]
+
+                            let cipherParams = ARTCrypto.defaultParamsWithKey(keyData, iv: ivData)
+                            let channelOptions = ARTChannelOptions(encrypted: cipherParams)
+                            let channel = client.channels.get("test", options: channelOptions)
+
+                            let transport = client.transport as! TestProxyTransport
+
+                            transport.beforeProcessingSentMessage = { protocolMessage in
+                                if protocolMessage.action == .Message {
+                                    expect(protocolMessage.messages![0].data as? String).to(equal(testMessage.encrypted.data))
+                                    expect(protocolMessage.messages![0].encoding).to(equal(testMessage.encrypted.encoding))
+                                }
+                            }
+
+                            transport.beforeProcessingReceivedMessage = { protocolMessage in
+                                if protocolMessage.action == .Message {
+                                    expect(protocolMessage.messages![0].data as? String).to(equal(testMessage.encrypted.data))
+                                    expect(protocolMessage.messages![0].encoding).to(equal(testMessage.encrypted.encoding))
+                                    // Force an error decoding a message
+                                    protocolMessage.messages![0].encoding = "bad_encoding_type"
+                                }
+                            }
+
+                            waitUntil(timeout: testTimeout) { done in
+                                let logTime = NSDate()
+
+                                channel.on(.Failed) { errorInfo in
+                                    expect(errorInfo).toNot(beNil())
+                                    expect(errorInfo).to(equal(channel.errorReason))
+                                }
+
+                                channel.subscribe(testMessage.encoded.name) { message in
+                                    expect(message.data as? String).to(equal(testMessage.encrypted.data))
+
+                                    let logs = querySyslog(forLogsAfter: logTime)
+                                    let line = logs.reduce("") { $0 + "; " + $1 } //Reduce in one line
+                                    expect(line).to(contain("ERROR: ARTDataDecoded failed to decode data as 'bad_encoding_type'"))
+
+                                    expect(channel.errorReason).toNot(beNil())
```

Comment:
Can we check that the error reason is the one we expected as opposed to a generic error reason?